### ISSUE TITLE
版本1.7.7.1的编译

### ISSUE THREAD

**Issue by anod221** [openresty/ngx_openresty#79]
文件：
ngx_http_upstream.h，第251-253行
代码：
```c++
#if (NGX_HTTP_GZIP)
    ngx_table_elt_t                 *content_encoding;
#endif
```
配置的时候，使用了--without-http_gzip_module，导致

文件：
ngx_http_redis_module.c，第565行
代码：
```c++
u->headers_in.content_encoding = h;
```
出现报错

**Comment by agentzh** [comment]
@anod221 The following patch fixes this issue, which will get included in the upcoming OpenResty 1.7.2.2 release:

https://github.com/openresty/ngx_openresty/blob/master/patches/ngx_http_redis-0.3.7-without_gzip.patch

I've also proposed the patch to the author of ngx_http_redis,  Sergey A. Osokin.

Thank you for the report!

**Comment by agentzh** [comment]
@anod221 FYI, the fix is already included in the OpenResty 1.7.2.2 RC2 pre-release:

http://openresty.org/download/ngx_openresty-1.7.7.2rc2.tar.gz
